lib/identity_cache/query_api.rb

```diff
@@ -154,6 +148,26 @@ def cache_fetch_includes
       end
     end
 
+    no_op_callback = proc {}
+    included do |base|
+      # Make sure there is at least once after_commit callback so that _run_commit_callbacks
+      # is called, which is overridden to do an early after_commit callback
+      base.after_commit(&no_op_callback)
```

Comment:
Why is this necessary? I believe `_run_commit_callbacks` gets called regardless of the number of callbacks.

Reply:
Looks like it isn't needed on rails master, but is needed for Active Record version 6.0.3.3.  I added this in response to seeing test failures.  There is some conditional logic around the call to `_run_commit_callbacks`, but I hadn't dug into why exactly this is needed.

Reply:
```suggestion
      after_commit(&proc {}) if _commit_callbacks.none?
```
You're right, we need this for previous versions we support.

Reply:
Active Record does in fact check to see if there are commit callbacks in [has_transactional_callbacks?](https://github.com/rails/rails/blob/v6.0.3.3/activerecord/lib/active_record/transactions.rb#L462-L464) which is a method that even exists on rails master.  `with_transaction_returning_status` uses that method in rails 6.0 to conditionally call `add_to_transaction` which is needed to make sure the commit callbacks are called.  On rails master, it unconditionally calls `add_to_transaction`, but uses `has_transactional_callbacks?` to determine the argument to pass into that method (`ensure_finalize`).